Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift

```diff
@@ -64,74 +83,174 @@ public struct LivestreamPlayer<Factory: ViewFactory>: View {
         onFullScreenStateChange: ((Bool) -> Void)? = nil
     ) {
         self.viewFactory = viewFactory
-        let viewModel = LivestreamPlayerViewModel(
-            type: type,
-            id: id,
-            muted: muted,
-            showParticipantCount: showParticipantCount
-        )
-        _viewModel = StateObject(wrappedValue: viewModel)
-        _state = StateObject(wrappedValue: viewModel.call.state)
+        let call = InjectedValues[\.streamVideo].call(callType: type, callId: id)
+        self.call = call
+        livestreamState = call.state.backstage ? .backstage : .live
+        countdown = 0
+        self.muted = muted
+        self.showParticipantCount = showParticipantCount
+        _state = ObservedObject(wrappedValue: call.state)
+        self.joinPolicy = joinPolicy
+        self.showsLeaveCallButton = showsLeaveCallButton
+        self.onFullScreenStateChange = onFullScreenStateChange
+        call.updateParticipantsSorting(with: livestreamOrAudioRoomSortPreset)
+    }
+    
+    internal init(
+        viewFactory: Factory = DefaultViewFactory.shared,
+        call: Call,
+        countdown: TimeInterval = 0,
+        livestreamState: LivestreamState? = nil,
+        muted: Bool = false,
+        showParticipantCount: Bool = true,
+        joinPolicy: JoinPolicy = .auto,
+        showsLeaveCallButton: Bool = false,
+        onFullScreenStateChange: ((Bool) -> Void)? = nil
+    ) {
+        self.viewFactory = viewFactory
+        self.call = call
+        self.countdown = countdown
+        self.livestreamState = livestreamState ?? (call.state.backstage ? .backstage : .live)
+        self.muted = muted
+        self.showParticipantCount = showParticipantCount
+        _state = ObservedObject(wrappedValue: call.state)
         self.joinPolicy = joinPolicy
         self.showsLeaveCallButton = showsLeaveCallButton
         self.onFullScreenStateChange = onFullScreenStateChange
-        viewModel.call.updateParticipantsSorting(with: livestreamOrAudioRoomSortPreset)
+        call.updateParticipantsSorting(with: livestreamOrAudioRoomSortPreset)
     }
 
     public var body: some View {
         ZStack {
-            if viewModel.errorShown {
+            if livestreamState == .error {
                 errorView
-            } else if viewModel.loading {
+            } else if livestreamState == .joining {
                 loadingView
-            } else if state.backstage {
+            } else if livestreamState == .backstage {
                 notStartedView
             } else {
                 videoRenderer
+                livestreamControls
             }
-            livestreamControls
         }
         .onChange(of: state.participants, perform: { newValue in
-            if viewModel.muted && newValue.first?.track != nil {
-                viewModel.muteLivestreamOnJoin()
+            if muted && newValue.first?.track != nil {
```

Comment:
what if the first participant is the pinned one and not the one actually having video track? shall we do `newValue.first { $0.hasVideo && $0.track != nil }` instead?